testing-unit-cs-top-level-program/README.md

```diff
@@ -0,0 +1,78 @@
+# C# Unit Testing with Top-Level Statements
+
+In this example project, we examine how to unit test C# Pulumi programs that are using top-level statements with dotnet SDK v6. These programs look like this
+
+```csharp
+using Pulumi;
+
+return await Deployment.RunAsync(() => 
+{
+    // Create resources here
+});
+```
+
+In order to unit test this piece of code, we have to do a bit of rewriting such that we don't run the `Deployment.RunAsync` function because it expects the program is being executed from the Pulumi CLI. 
+
+Now, we take the lambda that is provided to `Deployment.RunAsync` and move it in a place where we can reference it later from our unit test project. The following snippet shows this
+
+```csharp
+using Pulumi;
+
+return await Deployment.RunAsync(Deploy.Infra);
+
+class Deploy
```

Comment:
Does this need to be in a class? Could we reference a top level static method elsewhere?

Reply:
I tried not putting the code in a class 😞 but the `public` modifier is not allowed on top-level functions. I believe because their class is compiler-generated. Then I tried using reflection to get the method info and call it but I couldn't retrieve it. I think it is possible but it could messy so for now, a static method in a class is one way to go